### PR TITLE
docs: add handling errors for OpenAI provider

### LINKED ISSUE #499: docs(guides): fix error handling code for openAI completion
In [Use the Completion API](https://sdk.vercel.ai/docs/guides/providers/openai#use-the-completion-api), the code with error handling:

```ts
// ...

  // Ask OpenAI for a streaming completion given the prompt
  const response = await openai.completions.create({
    model: 'text-davinci-003',
    stream: true,
    prompt
  })
 
  // Check for errors -----> here
  if (!response.ok) {
    return new Response(await response.text(), {
      status: response.status
    })
  }

// ...
```

The `response` does not return a type of `Response`, therefore the code is not valid.

### Suggestion

```ts
// ...

  // Ask OpenAI for a streaming completion given the prompt
  const completions = await openai.completions.create({
    model: 'text-davinci-003',
    stream: true,
    prompt
  })

  // Convert the response into a friendly text-stream
  const stream = OpenAIStream(completions)

  // Respond with the stream
  const response = new StreamingTextResponse(stream)

  // Check for errors
  if (!response.ok) {
    return new Response(await response.text(), {
      status: response.status
    })
  }

  return response

// ...
```

### Reference

- [Use the Completion API](https://sdk.vercel.ai/docs/guides/providers/openai#use-the-completion-api)

### PR DESCRIPTION
**Pull request by devjiwonchoi**
> The response does not return a type of Response, therefore the code is not valid.

Fixes #499

### Reference

- [Use the Completion API](https://sdk.vercel.ai/docs/guides/providers/openai#use-the-completion-api)

### CODE REVIEW

**Comment by devjiwonchoi** [comment]
Absolutely! I’ll go over the docs for other error handlings also.

File: docs/pages/docs/guides/providers/openai.mdx [hunk 1]
```diff
@@ -217,6 +210,47 @@ export default function Completion() {
...
+    })
+
+    const stream = OpenAIStream(response)
+
+    return new StreamingTextResponse(stream)
+  } catch (error) {
+    // Check if the error is an APIError
+    if (error instanceof OpenAI.APIError) {
+      const { name, status, headers, message } = error
+      return NextResponse.json({ name, status, headers, message }, { status })
```
Comment (devjiwonchoi) [inline docs/pages/docs/guides/providers/openai.mdx:246]: @MaxLeiter Not sure if it was a good idea to add NextResponse here.

Comment (devjiwonchoi) [reply docs/pages/docs/guides/providers/openai.mdx:246]: Yes because it's Next specific, but if it's alright for you then I'm good with it.

Comment (MaxLeiter) [reply docs/pages/docs/guides/providers/openai.mdx:246]: Yeah let's use `new Response()` with JSON.stringify. Thanks for pointing this out!

**Comment by MaxLeiter** [comment]
thank you!

File: docs/pages/docs/guides/providers/openai.mdx (hunk 1 above)
Comment (devjiwonchoi) [reply docs/pages/docs/guides/providers/openai.mdx:246]: Um didn't resolve this yet but is it all good for now? Or I can fix it ASAP.

**Pull request merged**